src/TicketBai/Mews.Fiscalizations.TicketBai/ModelToDtoConverter.cs

```diff
@@ -0,0 +1,395 @@
+                IssuerType.IssuedByThirdParty, _ => EmitidaPorTercerosType.T,
+                IssuerType.IssuedByTransactionRecipient, _ => EmitidaPorTercerosType.D
+            );
+        }
+
+        private static IDDestinatario Convert(Receiver receiver)
+        {
+            var name = receiver.Match(
+                local => local.Name,
+                foreign => foreign.Name
+            );
+            var address = receiver.Match(
+                local => local.Address,
+                foreign => foreign.Address
+            );
+            var postalCode = receiver.Match(
+                local => local.PostalCode,
+                foreign => foreign.PostalCode
+            );
+            return new IDDestinatario
+            {
+                ApellidosNombreRazonSocial = name.Value,
+                CodigoPostal = postalCode.Value,
+                Direccion = address.Value,
+                Item = receiver.Match(
+                    local => (object)local.TaxpayerIdentificationNumber.TaxpayerNumber,
+                    foreign => new IDOtro1
+                    {
+                        CodigoPais = foreign.Country.Map(c => CovertCountryType21(c)).ToNullable(),
+                        CodigoPaisSpecified = foreign.Country.NonEmpty,
+                        ID = foreign.Id.Value,
+                        IDType = Convert1(foreign.IdType)
+                    }
+                )
+            };
+        }
+
+        private static HuellaTBAI1 Convert(InvoiceFooter footer)
+        {
+            return new HuellaTBAI1
+            {
+                EncadenamientoFacturaAnterior = footer.OriginalInvoiceInfo.Map(i => Convert(i)).GetOrNull(),
+                Software = Convert(footer.Software),
+                NumSerieDispositivo = footer.DeviceSerialNumber.Map(n => n.Value).GetOrNull()
+            };
+        }
+
+        private static EncadenamientoFacturaAnteriorType Convert(OriginalInvoiceInfo originalInvoiceInfo)
+        {
+            return new EncadenamientoFacturaAnteriorType
+            {
+                SerieFacturaAnterior = originalInvoiceInfo.Series.Map(s => s.Value).GetOrNull(),
+                NumFacturaAnterior = originalInvoiceInfo.Number.Value,
+                FechaExpedicionFacturaAnterior = Convert(originalInvoiceInfo.IssueDate),
+                SignatureValueFirmaFacturaAnterior = originalInvoiceInfo.Signature.Value
+            };
+        }
+
+        private static SoftwareFacturacionType1 Convert(Software software)
+        {
+            return new SoftwareFacturacionType1
+            {
+                LicenciaTBAI = software.License.Value,
+                Version = software.Version.Value,
+                Nombre = software.Name.Value,
+                EntidadDesarrolladora = Convert(software.Developer)
+            };
+        }
+
+        private static EntidadDesarrolladoraType1 Convert(Developer developer)
+        {
+            return developer.Match(
```

Comment:
Developer is exactly the same thing as Receiver. We don't even need a class for it, right?